topics/request.md

```diff
@@ -1,316 +1,116 @@
+## Parallel requests {id="parallel_requests"}
 
+When sending two requests at once, the client suspends the second request execution until the first is finished. If you need to perform several requests at once, you can use [launch](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/launch.html) or [async](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/async.html) functions. The code snippet below shows how to perform two requests asynchronously:
 ```kotlin
-data class HelloWorld(val hello: String)
-
-val client = HttpClient(Apache) {
-    install(JsonFeature) {
-        serializer = GsonSerializer {
-            // Configurable .GsonBuilder
-            serializeNulls()
-            disableHtmlEscaping()
-        }
-    }
-}
-
-client.post<Unit> {
-    url("http://127.0.0.1:8080/")
-    body = HelloWorld(hello = "world")
-}
 ```
+{src="snippets/client-parallel-requests/src/main/kotlin/com/example/Application.kt" lines="12,19-22"}
 
-Alternatively (using the integrated `JsonSerializer`):
+To see a full example, go to [client-parallel-requests](https://github.com/ktorio/ktor-documentation/tree/main/codeSnippets/snippets/client-parallel-requests).
 
-```kotlin
-val json = io.ktor.client.features.json.defaultSerializer()
-client.post<Unit>() {
-    url("http://127.0.0.1:8080/")
-    body = json.write(HelloWorld(hello = "world")) // Generates an OutgoingContent
-}
-```
 
-Or using Jackson (JVM only):
+## Cancel a request {id="cancel-request"}
 
+If you need to cancel a request, you can cancel a coroutine that runs this request. For instance, the [launch](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/launch.html) function returns a `Job` that can be used to cancel the running coroutine:
 ```kotlin
-val json = jacksonObjectMapper()
-client.post<Unit> {
-    url("http://127.0.0.1:8080/")
-    body = TextContent(json.writeValueAsString(userData), contentType = ContentType.Application.Json)
-}
 ```
+{src="snippets/_misc_client/CancelRequest.kt"}
 
->Remember that your classes must be *top-level* to be recognized by `Gson`. \\
->If you try to send a class that is inside a function, the feature will send a *null*.
->
-{type="note"}
-
-## Uploading multipart/form-data
-{id="multipart-form-data"}
-
-Ktor HTTP Client has support for making MultiPart requests.
-The idea is to use the `MultiPartFormDataContent(parts: List<PartData>)` as `OutgoingContent` for the body of the request.
-
-The easiest way is to use the [`submitFormWithBinaryData` method](#submit-form).
-
-Alternatively, you can set the body directly:
-
-```kotlin
-val request = client.request {
-    method = HttpMethod.Post
-    body = MultiPartFormDataContent(formData {
-        append("key", "value")
-    })
-}
-```
+Learn more from the [Cancellation and timeouts](https://kotlinlang.org/docs/cancellation-and-timeouts.html) help topic.
```

Comment:
Let's try and avoid using the construct of "help topic". We can just say "Learn more about Cancelations and timeouts"

Reply:
ok